tests/backend/test_rpc_handshake.py

```diff
@@ -0,0 +1,243 @@
+        assert rep.headers["Api-Version"] == str(API_VERSION)
+
+
+async def _test_handshake_api_version_header(
+    client: Union[AuthenticatedRpcApiClient, AnonymousRpcApiClient]
+):
+    server_versions = [ApiVersion(2, 1), ApiVersion(3, 1), ApiVersion(4, 1)]
+
+    with patch("parsec.backend.asgi.rpc.SUPPORTED_API_VERSIONS", server_versions):
+
+        # Plain invalide header value
+        rep = await client.send_ping(
+            extra_headers={"Api-Version": "dummy"},
+            check_rep=False,
+        )
+        assert rep.status_code == 422
+        assert rep.headers["Supported-Api-Versions"] == "2.1;3.1;4.1"
+
+        # Missing header, fallback to default value for backward compatibility
+        rep = await client.send_ping(
+            extra_headers={"Api-Version": None},
+            check_rep=False,
+        )
+        # Special case for anynous
+        assert rep.status_code == 200
+        rep.headers["Api-Version"] == "3.0"
+
+        # Client provides an incompatible api version
+        unknown_major = ApiVersion(version=5, revision=0)
+        too_old_major = ApiVersion(version=1, revision=0)
+        for bad_version in (too_old_major, unknown_major):
+            rep = await client.send_ping(
+                extra_headers={"Api-Version": str(bad_version)},
+                check_rep=False,
+            )
+            assert rep.status_code == 422
+            assert rep.headers["Supported-Api-Versions"] == "2.1;3.1;4.1"
+
+        # Client provides a compatible api version
+        unknown_minor = ApiVersion(version=4, revision=2)
+        rep = await client.send_ping(
+            extra_headers={"Api-Version": str(unknown_minor)},
+            check_rep=False,
+        )
+        assert rep.status_code == 200
+        assert rep.headers["Api-Version"] == "4.1"
+
+
+async def _test_handshake_content_type_header(
+    client: Union[AuthenticatedRpcApiClient, AnonymousRpcApiClient]
+):
+    # Bad header value
+    rep = await client.send_ping(
+        extra_headers={"Content-Type": "application/json"},
+        check_rep=False,
+    )
+    assert rep.status_code == 415
+    assert rep.headers["Api-Version"] == str(API_VERSION)
+
+    # Missing header, fallback to default value for backward compatibility
+    rep = await client.send_ping(
+        extra_headers={"Content-Type": None},
+        check_rep=False,
+    )
+    assert rep.status_code == 200
+    rep.headers["Content-Type"] == "application/msgpack"
+
+
+async def _test_authenticated_handshake_bad_signature_header(
+    client: AuthenticatedRpcApiClient, alice: LocalDevice, bob: LocalDevice
+):
+    # First test missing & plain bad headers
+    for expected_status_code, extra_headers in [
+        (401, {"Timestamp": None}),  # Missing Timestamp header
+        (401, {"Timestamp": "dummy"}),  # Bad Timestamp header
+        (401, {"Signature": None}),  # Missing Signature header
+        (401, {"Signature": "dummy"}),  # Bad Signature header
```

Comment:
```suggestion
        (401, {"Signature": b64encode(d"dummy")}),  # Bad Signature header
```

Reply:
the point is to have a plain wrong header, but also testing "half wrong" (with valid base64 but invalid content) is a good idea ^^

I've added this testcase